plugins/CoreHome/vue/src/ReportingPage/ReportingPage.vue

```diff
@@ -276,16 +276,16 @@ export default defineComponent({
           method: 'Live.getLastVisitsDetails',
           filter_limit: 1,
           doNotFetchActions: 1,
-        });
-      }).then((lastVisits) => {
-        if (!lastVisits || lastVisits.length === 0) {
-          this.hasRawData = false;
-          hideOnlyRawDataNoticifation();
-          return;
-        }
+        }).then((lastVisits) => {
```

Comment:
@sgiehl This change shouldn't make a difference, returning a promise in .then() will resolve the promise before going to the next chained then(). Can you check what `lastVisits` is set to both before this change and after it?

Reply:
Ah ok. Wasn't aware of that actually. But what happens if `undefined` is returned? Will the next then be skipped, called with `undefined` or with the data returned by the previous promise?

Reply:
@sgiehl the value for the next then() will be `undefined`. Whatever is returned from the previous then() is supplied to the next and if it's a promise its resolved first. If there is no explicit return, the function will implicitly return undefined which will be forwarded to the next then(). I'm also not 100% sure about this, but if a catch is invoked and returns a value instead of rethrowing, the value will be used in the next then().

Reply:
But if `undefined` is passed to the next `then`, I guess it's correct to move it. if `this.hasRawData` was set to true in a previous check and the check runs again, the first `then` will execute this part:
https://github.com/matomo-org/matomo/blob/6ab9923d6324c3d98765881e525d7ecf8ba9d831/plugins/CoreHome/vue/src/ReportingPage/ReportingPage.vue#L270-L273
the chained `then` will receive `undefined` causing this to be executed:
https://github.com/matomo-org/matomo/blob/6ab9923d6324c3d98765881e525d7ecf8ba9d831/plugins/CoreHome/vue/src/ReportingPage/ReportingPage.vue#L281-L285

So the raw data message would be hidden again, even though it should actually be shown. Or do I understand something wrong there?

Reply:
@sgiehl yes, good point, that u derstanding is correct, that was not obvious to me at first glance. The change makes sense.